mergify_engine/queue/merge_train.py

```diff
@@ -812,6 +815,20 @@ async def _remove_duplicate_pulls(self) -> None:
                 wp_to_keep.append(wp)
         self._waiting_pulls = wp_to_keep
 
+    async def _sync_configuration_change(self, queue_rules: rules.QueueRules) -> None:
+        for i, pseudo_car in enumerate(list(self._iter_pseudo_cars())):
+            queue_rule = queue_rules.get(pseudo_car.config["name"])
+            if queue_rule is None:
+                # NOTE(sileht): We just slice the cars list here, so when the
+                # car will be recreated if the rule doesn't exists anymore, the
+                # failure will be reported properly
+                await self._slice_cars_at(i)
+            else:
+                # NOTE(sileht): Update the queue config, so if
+                # speculative_checks change the train will grow or shrink
+                # instantly
+                pseudo_car.config["queue_config"] = queue_rule.config
```

Comment:
Do we actually want this, rather than reading the config from the config file?

Reply:
A long time ago this was not possible as we didn't have the configuration everywhere we need it. This may be possible now.
But that not a easy refactoring, `current` values has to be stored for the comparaison. So the config will move from per Car to per Train. I keep MRGFY-522 open to do it.